File: src/test/java/com/study/platform/support/fake/FakeTokenRepository.java
package com.study.platform.support.fake;

import com.study.platform.global.auth.port.TokenRepository;

import java.time.Duration;
import java.util.HashMap;
import java.util.Map;
import java.util.Optional;

public class FakeTokenRepository implements TokenRepository {

    private final Map<String, String> store = new HashMap<>();

    @Override
    public void save(String key, String value, Duration ttl) {
        store.put(key, value);
    }

    @Override
    public Optional<String> find(String key) {
        return Optional.ofNullable(store.get(key));
    }

    @Override
    public void delete(String key) {
        store.remove(key);
    }
}

File: domain/src/testFixtures/kotlin/com/study/platform/support/fake/FakeTokenRepository.kt
package com.study.platform.support.fake

import com.study.platform.global.auth.port.TokenRepository
import java.time.Duration

class FakeTokenRepository : TokenRepository {

    private val store: MutableMap<String, String> = HashMap()

    override fun save(key: String, value: String, ttl: Duration) {
        store[key] = value
    }

    override fun find(key: String): String? = store[key]

    override fun delete(key: String) {
        store.remove(key)
    }
}
